src/daemon/Greeter.cpp

```diff
@@ -197,7 +197,7 @@ namespace SDDM {
             env.insert(QStringLiteral("XDG_SEAT"), m_display->seat()->name());
             env.insert(QStringLiteral("XDG_SEAT_PATH"), daemonApp->displayManager()->seatPath(m_display->seat()->name()));
             env.insert(QStringLiteral("XDG_SESSION_PATH"), daemonApp->displayManager()->sessionPath(QStringLiteral("Session%1").arg(daemonApp->newSessionId())));
-            if (m_display->seat()->name() == QLatin1String("seat0"))
+            if (m_display->seat()->name() == QLatin1String("seat0") && m_display->seat()->canTty())
```

Comment:
IMO should check for `m_display->terminalId() > 0`

Reply:
Done locally